### PR TITLE
feat: expose parser options as a seperate type

### LINKED ISSUE #345: Expose parser options in separate type
#### Reason/Context
The parser options is current an inline definition https://github.com/asyncapi/parser-js/blob/aba3ed25a58e1eb36b77d5e7dd724202ca737f64/lib/parser.js#L33

This however also means that the types are generated inline, making reusability impossible https://github.com/asyncapi/parser-js/blob/aba3ed25a58e1eb36b77d5e7dd724202ca737f64/types.d.ts#L1077

#### Description
We need to change it to have a separate type, so it is generated separately from the function itself. JSDoc allows something like [typedef](https://jsdoc.app/tags-typedef.html), which even though we don't reuse the options internally, externally would be helpful.

### PR DESCRIPTION
**Pull request by iamdevelopergirl**
**Description**

- Exposing the options argument in the parser function as a separate type, so that it can be reused.

**Related issue(s)**
Fixes #345

### CODE REVIEW

File: lib/parser.js [hunk 1]
```diff
@@ -118,7 +123,7 @@ async function parse(asyncapiYAMLorJSON, options = {}) {
  * 
  * @param {String} url URL where the AsyncAPI document is located.
  * @param {Object} [fetchOptions] Configuration to pass to the {@link https://developer.mozilla.org/en-US/docs/Web/API/Request|fetch} call.
- * @param {Object} [options] Configuration to pass to the {@link module:Parser#parse} method.
+ * @param {Object} [options] Configuration to pass to the {@link ParserOptions} method.
```
Comment (magicmatatjahu) [inline lib/parser.js:126]:
```suggestion
 * @param {ParserOptions} [options] Configuration to pass to the {@link ParserOptions} method.
```

File: lib/parser.js [hunk 2]
```diff
@@ -243,7 +248,7 @@ function applyTraits(js) {
  * @param  {Object} parsedJSON parsed AsyncAPI document
  * @param  {String} asyncapiYAMLorJSON AsyncAPI document in string
  * @param  {String} initialFormat information of the document was originally JSON or YAML
- * @param  {Object} options Configuration options.
+ * @param  {Object} options Configuration options. {@link ParserOptions}
```
Comment (magicmatatjahu) [inline lib/parser.js:251]:
```suggestion
 * @param  {ParserOptions} options Configuration options. {@link ParserOptions}
```

File: lib/parser.js [hunk 3]
```diff
@@ -274,7 +279,7 @@ async function customChannelsOperations(parsedJSON, asyncapiYAMLorJSON, initialF
...
```
Comment (magicmatatjahu) [inline lib/parser.js:282]:
```suggestion
 * @param  {ParserOptions} options Configuration options. {@link ParserOptions}
```

**Comment by magicmatatjahu** [comment]
@iamdevelopergirl Hello! Sorry for late review, could you add my suggestions? :)

**Review by iamdevelopergirl — COMMENTED** [review]
Added the suggestions to change the type of options

**Comment by iamdevelopergirl** [comment]
Thanks for making the changes. Guess I overlooked the type of options param. :)

**Comment by magicmatatjahu** [comment]
@iamdevelopergirl Sorry, I don't see them. Did you push them or do you have only in the local repo? 😅

**Comment by iamdevelopergirl** [comment]
@magicmatatjahu Right, you got me :)
Added it now.

**Comment by magicmatatjahu** [comment]
@iamdevelopergirl No worries :) If I could, last thing. Could you rebase your repo with base (master) branch? I don't have permission to do that by myself (and it blocks me to merge), so you must do it in your side.

**Comment by iamdevelopergirl** [comment]
Yep. Done 👍

**Review by magicmatatjahu — APPROVED** [review]

**Comment by magicmatatjahu** [comment]
@iamdevelopergirl Thanks for contribution and patience 😅

**Comment by magicmatatjahu** [comment]
@all-contributors please add @iamdevelopergirl for code

**Comment by iamdevelopergirl** [comment]
Cool. Thanks

**Pull request merged**